lib/msal-core/src/AuthError.ts

```diff
@@ -0,0 +1,240 @@
+/**
+  * Copyright (c) Microsoft Corporation
+  *  All Rights Reserved
+  *  MIT License
+  *
+  * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  * software and associated documentation files (the "Software"), to deal in the Software
+  * without restriction, including without limitation the rights to use, copy, modify,
+  * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  * permit persons to whom the Software is furnished to do so, subject to the following
+  * conditions:
+  *
+  * The above copyright notice and this permission notice shall be
+  * included in all copies or substantial portions of the Software.
+  *
+  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+  * OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
+  * OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+  */
+
+import { Constants } from "./Constants";
+import { ErrorMessage } from "./ErrorMessage";
+import { Utils } from "./Utils";
+/**
+* @hidden
+*
+* General error class thrown by the MSAL.js library.
+*/
+export class AuthError extends Error {
+    error: string;
+    errorDesc: string;
+
+    constructor(error: string, detailMessage: string) {
+        super(detailMessage);
+        this.error = error;
+        this.errorDesc = detailMessage;
+        this.name = "AuthError";
+        this.stack = new Error().stack;
+
+        Object.setPrototypeOf(this, AuthError.prototype);
```

Comment:
what does this do ?

Reply:
like what does in accomplish in usage?

Reply:
This enabled us to extend the JS built-in Error object. We have to do this because inheritance for built-ins is a little bit more convoluted in Typescript 2.2+. https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
+1 for interesting weird stuff ... haha it's kinda gross but not your fault